### DISCUSSION TITLE
[bug]: new version of CLN and maybe eclair won't force close channel from SCB request

### DISCUSSION THREAD

**Discussion by btweenthebars** [lightningnetwork/lnd#7465] [Troubleshooting] [answered]
### Background

cdecker has it that there's a change in the bolt spec on the work flow of SCB that hasn't been implemented on LND

Please take a look [here](https://github.com/ElementsProject/lightning/issues/5876) . 

### Your environment

v0.15.5-beta

### Steps to reproduce

I mentioned my incident in [here](https://github.com/lightningnetwork/lnd/issues/7273#issuecomment-1372435789) as I wasn't sure what's going on

**Comment by Roasbeef** [comment]
So we investigated a bit, and concluded that we were already sending the `Error` once the SCB flow was triggered. So instead it seems some other behavior changed here that is preventing some other implementations from closing either once we send that `Error`, or just from receiving the purposefully incorrect sync message.

**Comment by Crypt-iQ** [comment]
If you are experiencing this issue on `v0.15.5-beta`, the following steps should resolve the issue:

- Ensure that you have already tried connecting to the CLN node and received `Warning` followed by a disconnect.
- Checkout https://github.com/Crypt-iQ/lnd/tree/scb_cln_issue_7301_v0.15.5-beta
- Build `lnd` with the `dev` build tag.
- Boot up the newer version of `lnd` with the flag `--protocol.custom-message=17`.
- If you have logs, try to fetch the 32-byte (64 hexadecimal characters) `chan_id`/`ChannelID`/`channel_id` value. This may be appear with `ChannelLink` logs in the `HSWC` category. If you can't fetch it, feel free to post a comment asking for help.
- Once you have the 32-byte `chan_id`/`ChannelID`/`channel_id` value, fetch the peer's hexadecimal pubkey. This should appear in `listpeers` if you are connected to the peer or in some of your logs.
- Finally, use the `sendcustom` API like so:
```
lncli sendcustom --peer=<peer pubkey in hex> --type=17 --data=`chan_id`0000
```
- Note that the `data` field in the `sendcustom` call is the `chan_id` followed by 4 zeros.
- Once you see that your channel has been force closed via the `lnd` API, continue using the original `v0.15.5-beta`. This patch is only meant to be used temporarily.

If you are uncomfortable with using my personal branch and would instead prefer a temporary release, please let us know. Or if you don't know how to perform any of the steps above, leave a comment and I can help out.

**Comment by Crypt-iQ** [comment]
@btweenthebars were you able to resolve your issue using the steps above?

**Comment by btweenthebars** [comment]
> @btweenthebars were you able to resolve your issue using the steps above?

Hi sorry, I'm on CLN ( the latest version) not LND

**Comment by Crypt-iQ** [comment]
Can you contact your peer to see if he wants to use the steps to above?

**Comment by btweenthebars** [comment]
> Can you contact your peer to see if he wants to use the steps to above?

Well, that's a while ago and I already manually closed the channels. Next time I see LND peers with the issue, I'll tell them to try the steps above.

**Comment by ardevd** [comment]
I'm struggling with this bug. I've build @Crypt-iQ 's lnd version, but I can't seem to find the channel ID. Nothing in the logs with the HSWC tag or any mention of CahnnelLink anywhere.

**Comment by Crypt-iQ** [comment]
> I'm struggling with this bug. I've build @Crypt-iQ 's lnd version, but I can't seem to find the channel ID. Nothing in the logs with the HSWC tag or any mention of CahnnelLink anywhere.

What's the funding point of the channel?

**Comment by ardevd** [comment]
> > I'm struggling with this bug. I've build @Crypt-iQ 's lnd version, but I can't seem to find the channel ID. Nothing in the logs with the HSWC tag or any mention of CahnnelLink anywhere.
> 
> What's the funding point of the channel?

`d51881f36667cc318d5309bb4699254e26d7360a6c9603003c3c649e1b6ef76d:1`

**Comment by Crypt-iQ** [comment]
> d51881f36667cc318d5309bb4699254e26d7360a6c9603003c3c649e1b6ef76d:1

your channel id to use in the command should be `d51881f36667cc318d5309bb4699254e26d7360a6c9603003c3c649e1b6ef76c`

**Comment by ardevd** [comment]
Thanks. Care to share how it's calculated? why was `f76d` changed to `f76c`?

**Comment by guggero** [comment]
Now that I saw the example, I remember... I think it's XOR of the last byte (`0x6d`) and the outpoint index (`0x01`) in this case. So if the outpoint index (the part after the colon) is `0`, then the channel ID is equal to the TXID (the part before the colon), if it's something else, you need to XOR.

**Comment by ardevd** [comment]
Btw, @Crypt-iQ your `scb_cln_issue_7301_v0.15.5-beta` branch doesnt seem to contain support for `protocol.custom-message`. I had to use the ``scb_cln_issue_7301` branch.

**Comment by ardevd** [comment]
@Crypt-iQ is this the expected log output after sending the custom message? 

```
2023-02-23 18:53:33.140 [DBG] PEER: Peer(0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754): Sending Error(type=17) to 0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754@198.58.118.99:9735
2023-02-23 18:53:33.301 [DBG] PEER: Peer(0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754): Received Error(chan_id=89b72329e1c276633de91fed43ffc2a233f7a6bf503bb0b60ae65929dc797d93, err=Unknown channel for WIRE_ERROR) from 0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754@198.58.118.99:9735
2023-02-23 18:53:33.301 [INF] PEER: Peer(0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754): unable to read message from peer: EOF
2023-02-23 18:53:33.301 [INF] PEER: Peer(0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754): disconnecting 0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754@198.58.118.99:9735, reason: read handler closed
2023-02-23 18:53:33.301 [INF] NTFN: Cancelling epoch notification, epoch_id=10
2023-02-23 18:53:33.401 [DBG] SRVR: Peer 0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754@198.58.118.99:9735 has been disconnected
2023-02-23 18:53:33.402 [DBG] FNDG: Cancelling all reservations for peer 0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754
2023-02-23 18:53:33.402 [DBG] FNDG: No active reservations for node: 0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754
2023-02-23 18:53:33.402 [INF] DISC: Removing GossipSyncer for peer=0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754
2023-02-23 18:53:33.402 [DBG] SRVR: Notifying that peer 0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754@198.58.118.99:9735 is offline
2023-02-23 18:53:33.402 [DBG] SRVR: removing peer 0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754@198.58.118.99:9735
2023-02-23 18:53:33.402 [DBG] PRNF: PeerNotifier notifying peer: 0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754 offline
2023-02-23 18:53:33.402 [DBG] BTCN: Disconnected from 0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754@198.58.118.99:9735 (reqid 4)
2023-02-23 18:53:33.402 [DBG] DISC: Stopping GossipSyncer(0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754)
```

**Comment by Crypt-iQ** [comment]
> Btw, @Crypt-iQ your scb_cln_issue_7301_v0.15.5-beta branch doesnt seem to contain support for protocol.custom-message. I had to use the ``scb_cln_issue_7301` branch.

Ah that's right, were you on v0.15.5-beta before?

> 2023-02-23 18:53:33.301 [DBG] PEER: Peer(0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754): Received Error(chan_id=89b72329e1c276633de91fed43ffc2a233f7a6bf503bb0b60ae65929dc797d93, err=Unknown channel for WIRE_ERROR) from 0317983322379d859c0d43a90c8dcd3e7239b8e0671b00a657ce3924d4498f3754@198.58.118.99:9735

That is not the expected output. That means CLN doesn't recognize the channel. This `chan_id` is also wrong. Can you paste the entire command of `sendcustom`?

**Comment by ardevd** [comment]
Yeah, that was for another channel. Apologies for the confusion.

Here is the one I discussed above:
`./lncli sendcustom --peer=0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f --type=17 --data=d51881f36667cc318d5309bb4699254e26d7360a6c9603003c3c649e1b6ef76c0000`

And output:

```
2023-02-23 19:01:48.008 [DBG] PEER: Peer(0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f): Sending Error(type=17) to 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f@73.114.40.246:9735
2023-02-23 19:01:48.029 [DBG] BTCN: Attempting to connect to 34.73.79.114:8333 (reqid 12)
2023-02-23 19:01:48.100 [DBG] SRVR: Broadcasting 1 messages
2023-02-23 19:01:48.100 [DBG] SRVR: Broadcasting 1 messages
2023-02-23 19:01:48.100 [DBG] SRVR: Broadcasting 1 messages
2023-02-23 19:01:48.100 [DBG] SRVR: Broadcasting 1 messages
2023-02-23 19:01:48.100 [DBG] SRVR: Broadcasting 1 messages
2023-02-23 19:01:48.100 [DBG] SRVR: Broadcasting 1 messages
2023-02-23 19:01:48.100 [DBG] SRVR: Broadcasting 1 messages
2023-02-23 19:01:48.100 [DBG] SRVR: Broadcasting 1 messages
2023-02-23 19:01:48.100 [DBG] SRVR: Broadcasting 1 messages
2023-02-23 19:01:48.100 [DBG] SRVR: Broadcasting 1 messages
2023-02-23 19:01:48.132 [DBG] PEER: Peer(0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f): Received Error(chan_id=d51881f36667cc318d5309bb4699254e26d7360a6c9603003c3c649e1b6ef76c, err=Unknown channel for WIRE_ERROR) from 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f@73.114.40.246:9735
2023-02-23 19:01:48.132 [INF] PEER: Peer(0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f): unable to read message from peer: EOF
2023-02-23 19:01:48.133 [INF] PEER: Peer(0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f): disconnecting 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f@73.114.40.246:9735, reason: read handler closed
2023-02-23 19:01:48.133 [INF] NTFN: Cancelling epoch notification, epoch_id=15
2023-02-23 19:01:48.143 [DBG] BTCN: Connected to 34.73.79.114:8333 (reqid 12)
2023-02-23 19:01:48.233 [DBG] SRVR: Peer 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f@73.114.40.246:9735 has been disconnected
2023-02-23 19:01:48.233 [DBG] FNDG: Cancelling all reservations for peer 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f
2023-02-23 19:01:48.233 [DBG] FNDG: No active reservations for node: 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f
2023-02-23 19:01:48.233 [INF] DISC: Removing GossipSyncer for peer=0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f
2023-02-23 19:01:48.233 [DBG] SRVR: Notifying that peer 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f@73.114.40.246:9735 is offline
2023-02-23 19:01:48.233 [DBG] SRVR: removing peer 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f@73.114.40.246:9735
2023-02-23 19:01:48.233 [DBG] PRNF: PeerNotifier notifying peer: 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f offline
2023-02-23 19:01:48.233 [DBG] DISC: Stopping GossipSyncer(0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f)
2023-02-23 19:01:48.233 [DBG] BTCN: Disconnected from 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f@73.114.40.246:9735 (reqid 11)
2023-02-23 19:01:48.233 [DBG] SRVR: Scheduling connection re-establishment to persistent peer 0298f6074a454a1f5345cb2a7c6f9fce206cd0bf675d177cdbf0ca7508dd28852f in 2.1s
```

**Comment by ardevd** [comment]
> > Btw, @Crypt-iQ your scb_cln_issue_7301_v0.15.5-beta branch doesnt seem to contain support for protocol.custom-message. I had to use the ``scb_cln_issue_7301` branch.
> 
> Ah that's right, were you on v0.15.5-beta before?
> 

You wrote > Checkout https://github.com/Crypt-iQ/lnd/tree/scb_cln_issue_7301_v0.15.5-beta  

so I tried to build that tree first and found that it didnt have support for the `protocol.custom-message` argument. The `scb_cln_issue_7301` branch worked.

**Comment by ardevd** [comment]
@Crypt-iQ any idea why it's not working?

**Comment by guggero** [comment, marked as answer]
@ardevd I pushed up a new branch that builds on top of the just-tagged `v0.16.0-beta.rc1` and adds a command to make it easier to use: https://github.com/lightningnetwork/lnd/tree/scb_cln_issue_7301_v0.16.0-beta.rc1

Here's the updated HOWTO:
- Checkout https://github.com/lightningnetwork/lnd/tree/scb_cln_issue_7301_v0.16.0-beta.rc1 (`git clone https://github.com/lightningnetwork/lnd && cd lnd && git checkout scb_cln_issue_7301_v0.16.0-beta.rc1`)
- Build `lnd` with the `dev` build tag (`make install tags=dev`)
- Boot up the newer version of `lnd` with the flag `--protocol.custom-message=17`.
- Finally, use the `sendchannelerror` command like so:
```
lncli sendchannelerror --peer=<peer pubkey in hex> --chan_point <channel_point_with_colon_notation>
```
- Once you see that your channel has been force closed via the `lnd` API, continue using the original `v0.16.0-beta.rc1`. This patch is only meant to be used temporarily.

Could you please try with that one again, just to make sure the issue isn't the channel ID?

**Comment by ardevd** [comment]
Worked like a dream. Force close tx was broadcast to the mempool shortly after running the command.

**Comment by guggero** [comment]
Given that the solution above might be a problem for users running Umbrel and other node packaging solutions where running a custom version of `lnd` is not easy, I created a specific new command in [`chantools`](https://github.com/guggero/chantools) to solve this.

Steps (alternative to running the custom version of `lnd` mentioned above):
 - Download [the latest version of `chantools` (`v0.10.7`)](https://github.com/guggero/chantools/releases/tag/v0.10.7) (can be any machine, use the `darwin` release for MacOS).
 - Run the following command (replacing the peer address with the full address of the remote peer and the channel point with the channel in question):
```
chantools triggerforceclose --peer 039d9c531877266e190af2d51795b6104a1e91a188dad4b9ccf3ae4154075dd8e3@localhost:9747 --channel_point 8e3efb6f5537d964c1c02c0587f75fc6995888675083400d4b10c02f47ad3592:0
```
 - The tool **will ask you for your seed**, which is required to initiate the encrypted p2p communication with the peer. 
 - If the channel is observed by `lnd` (e.g. showing up in the list of `lncli pendingchannels`), then just wait until the force close transaction is confirmed and the channel should automatically update the status there.
 - If the channel isn't showing up in the list of pending channels, you can use the `chantools sweepremoteclosed` command to sweep the funds.

**Reply by wealthcollector** [reply]
I can't get this to work. The peer address is a tor address and when I run the command I get this: -bash: chantools: command not found

I'm running MacOs. After downloading, is there anything else I need to do before running the command. I have 3 channels stuck in waiting to close, but can't get them to force close. Thank you.

I tried with and without the colon for channel point 
<img width="758" alt="Screen Shot 2023-12-26 at 4 09 44 PM" src="https://github.com/lightningnetwork/lnd/assets/153935966/61cd7bd3-2bdd-4312-ae12-b0c1afcf8844">

**Reply by wealthcollector** [reply]
can you help please? @guggero

**Reply by saubyk** [reply]
Hi @wealthcollector are you on the right version of chantools?

**Reply by wealthcollector** [reply]
Hi @saubyk. I'm on a mac and I downloaded the darwin one from here https://github.com/lightninglabs/chantools/releases. After this, it went into my downloads folder and looks like this. 
<img width="473" alt="Screenshot 2024-01-03 at 8 19 42 PM" src="https://github.com/lightningnetwork/lnd/assets/153935966/077b7eae-06e8-4f3e-a87b-9b304f6e3efa">

After this, I ran the triggerforceclose command above that I pasted in terminal and got the  "-bash:  chantools: command not found" message. Not sure if i was supposed to do something else before running the command or not.

**Reply by saubyk** [reply]
might be a simple mistake. have you tried running the command with `./chantools...`?

**Reply by wealthcollector** [reply]
-bash: ./chantools: No such file or directory 

![Screenshot 2024-01-03 at 8 29 28 PM](https://github.com/lightningnetwork/lnd/assets/153935966/2ba70993-6b89-4ad7-9fc2-e4fd96f43d8f)

**Reply by saubyk** [reply]
couple of questions:
1. which platform is the umbrel running on? Mac or a raspberry pi?
2. did you follow the instructions (https://github.com/lightninglabs/chantools#installation) to install chantools?
3. what's the output of the command...`which chantools`?

**Reply by wealthcollector** [reply]
1. Umbrel is running on rasberry pi.
2. Just followed and will paste results of installation below
3. I got no output for that command

<img width="288" alt="Screenshot 2024-01-03 at 8 44 42 PM" src="https://github.com/lightningnetwork/lnd/assets/153935966/34f05181-e6b3-456b-86de-a334a785fe2f">
<img width="553" alt="Screenshot 2024-01-03 at 8 44 59 PM" src="https://github.com/lightningnetwork/lnd/assets/153935966/d521ea51-a081-47a6-9e87-a4dcf5c7ff38">
<img width="487" alt="Screenshot 2024-01-03 at 8 45 41 PM" src="https://github.com/lightningnetwork/lnd/assets/153935966/1253c5cc-4ba7-4196-87cb-3b3ec14fbf99">
<img width="362" alt="Screenshot 2024-01-03 at 8 46 47 PM" src="https://github.com/lightningnetwork/lnd/assets/153935966/fea9e36b-765b-47f4-87d5-30efc905f3bd">

**Reply by saubyk** [reply]
So, basically you need to install chantools on the device where your node is running, in this case your raspberry pi (not the Mac).

From the screenshots above, it appears that the `make` failed on your umbrel, because it doesn't have `go`. I would recommend installing the binary instead. Follow the instructions on the install page or below:
```
$ cd /tmp
$ wget -O chantools.tar.gz https://github.com/lightninglabs/chantools/releases/download/v0.12.0/chantools-linux-armv7-v0.12.0.tar.gz
$ tar -zxvf chantools.tar.gz
$ sudo mv chantools-*/chantools /usr/local/bin/
```
Once you've executed the above, try running `which` again to make sure that tool is installed

**Reply by wealthcollector** [reply]
That worked. I then ran chantool trigger force closed and it asked me the seed words like it's supposed to. I input them but it had an issue with the peer address. I think it's because its a tor address? Any idea what I should do with that? 

<img width="919" alt="Screenshot 2024-01-03 at 9 01 16 PM" src="https://github.com/lightningnetwork/lnd/assets/153935966/2a2bd8cc-98cf-4a8d-b081-1404f931425c">

**Reply by saubyk** [reply]
cc: @guggero if you have a response to the above

**Reply by wealthcollector** [reply]
Tried closing a different channel with an IPv4 address and got an "error connecting to peer: pper did not complete handshake within 15s"

<img width="1346" alt="Screenshot 2024-01-03 at 9 20 50 PM" src="https://github.com/lightningnetwork/lnd/assets/153935966/19dae9e5-0388-4d38-9459-278f4939115d">
<img width="1346" alt="Screenshot 2024-01-03 at 9 21 39 PM" src="https://github.com/lightningnetwork/lnd/assets/153935966/4dc1c47b-08da-4897-bc16-a41cca7a9987">

Then tried same command again and got this:

<img width="542" alt="Screenshot 2024-01-03 at 9 29 22 PM" src="https://github.com/lightningnetwork/lnd/assets/153935966/55beb113-476c-4198-a227-a8ca5d2bfcf3">

**Reply by guggero** [reply]
Tor addresses are not supported. So only ipv4 will work. And the error you got for the ipv4 address sounds like your peer has a connection issue.

But why are you attempting to run `triggerforceclose` in the first place? Do you not have an SCB file? The command will only work for very specific versions of CLN, so I doubt it's the right command for the situation.

**Reply by wealthcollector** [reply]
A storm damaged my SSD and I wiped it before saving the backup file (i know huge mistake). I did have a backup file that was a few months out of date that was saved. I got a bunch of people to close their channels with me on LN+, and got most of my funds back onchain, but I still needed to sweep some of my funds, so I had used umbrels "recover your channels" that had a backup file from my last state before the storm. I got those funds swept, but i still have 3 channels that are pending/waiting to close that I'm trying to run triggerforceclose so the channels will close and can get funds back on chain. Otherwise they're stuck in limbo right now and don't know what to do if my peers don't close them. I've messaged them on LN+, but they haven't been active in awhile. I'm a newb and technical abilities aren't that great, but have learned a lot and made a lot of mistakes with this that I hopefully won't do again when I get a new node up and running. I hope you can help and thanks for your time.

**Reply by guggero** [reply]
> but i still have 3 channels that are pending/waiting 

It could be that the peers already force closed but since the mempool is too full because of the fees they never confirmed.
If that is the case then `chantools pullanchor` (only available in the latest version, make sure to download `v0.12.1`) can help you. You can find out if there is a force close transaction in the mempool by going to mempool.space, and entering the first part of the channel point there, then follow the channel output.

**Reply by wealthcollector** [reply]
Actually 1 of the channels was closed by a guy who read his message yesterday on LN+ after I sent it a month ago, so only 2 channels left to go. I checked those 2 channels on mempool and it's showing that both are still open. One is tor and one is ipv4. So if chantools doesn't work, is my only other option to wait for them to close the channel with me?  The ipv4 channel that is opened with me is via a batched channel open that they did (maybe payjoin?). Would that be the reason chantools wouldn't work? Not sure i even have a local balance there and wouldn't make sense to close with them as it would close out their other 21 channel openings from the batch open? So i guess mostly looking to close the tor one and let any small losses with the batched channel go. Again, I appreciate you helping me.

**Reply by guggero** [reply]
Creating multiple channels in a batch should have no influence on whether you can use `chantools` or not. And from the error message it does just sound like a bad or very slow connection on their end. So yes, waiting or attempting to contact them somehow are probably your only options. They won't have to close all 21 channels, the channels are all independent, even if they were created in a single batch.

**Reply by guggero** [reply]
You could try the command with the ipv4 channel again, perhaps it's dependent on their node's load or available network bandwidth.

**Reply by wealthcollector** [reply]
Not sure if it's because of the new chantools update or what, but it finally worked with the command. The only issue is that the channel closure isn't showing up in the mempool with the transaction ID that was given. Not sure what to do now. Do I need to do the sweepremoteclosed command? Wouldn't the channel need to close first? Thanks.

2024-01-21 03:58:13.032 [INF] CHAN: Message sent, waiting for force close transaction to appear in mempool
2024-01-21 03:58:17.409 [INF] CHAN: Found force close transaction (i removed it for privacy)
2024-01-21 03:58:17.409 [INF] CHAN: You can now use the sweepremoteclosed command to sweep the funds from the channel

**Reply by wealthcollector** [reply]
@saubyk you able to help?

**Reply by guggero** [reply]
Could be an issue of the full mempool... But if the command gives you a TXID, it means it is in _some_ mempool. Do you see it on mempool.space for example? So just try the sweepremoteclosed command and give it a high-enough fee, then it should CPFP the force close connection as well.

**Reply by wealthcollector** [reply]
Thank you @guggero! For some reason, it wasn't showing up in my local mempool connected to my node. Maybe cause the fee was too low, but yes when i check mempool.space it shows it. I now have 1 channel left I'm trying to close. It's a tor channel and has been stuck in "waiting close" for months now on RTL. When I check my node on mempool, it shows that the channel is inactive, but still open and there is no closure TXID. Have tried lncli force close and get "bash: lncli: command not found". Have tried chantools but as you have mentioned it doesn't work for TOR.  Someone mentioned that the new chantools pull anchor command might work. Will this work for me? Don't want to mess anything up by trying it. I tried messaging the other party of this channel on LN+ 3 months ago to close the channel, but they haven't been on to check their messages. Any help to get this channel closed and back in my onchain wallet would be much aprreciated. Thanks again for your time and help.

**Reply by guggero** [reply]
If you get "command not found" it means something with your installation is not correct. Or you are in the wrong shell. Try the "Lightning Shell" of Umbrel.
Depending on the type of channel and the size of your node's mempool you might be able to force close the channel. If you paste the TXID (not going to type it from a screenshot) I can take a look at its state.

**Reply by wealthcollector** [reply]
txid: 87405ec63fc490563aaa7b5e2f420fa608ccbe6f7cef75cb0b79a3d9a9aa
81dc


https://mempool.space/lightning/channel/870047949272055809

**Reply by guggero** [reply]
Thanks. Yeah, there is nothing in the mempool yet, so you'll need to try to force close it yourself. The command to run is: `lncli closechannel 87405ec63fc490563aaa7b5e2f420fa608ccbe6f7cef75cb0b79a3d9a9aa81dc 1 --force`. If that doesn't cause the transaction to be found in mempool.space, you might need to increase your `bitcoind`'s mempool size by adding `maxmempool=2048` to the `bitcoin.conf` file.

**Reply by wealthcollector** [reply]
I got this when running the command:  [lncli] rpc error: code = Unknown desc = cannot close channel with state: ChanStatusRestored

**Reply by guggero** [reply]
Ah, right, I forgot that you restored from SCB, sorry. Then you cannot force close yourself. Unfortunately there is nothing else you can do other than wait and hope the peer eventually force closes and the transaction makes it into the mempool somehow.

**Reply by wealthcollector** [reply]
Damn :(. Ok one last question for you. Since I could be waiting indefinitely for them to close this channel, is it ok to transfer the entire balance of my onchain wallet to cold storage or do I need to keep some amount on here in the event of a channel closure? I've read some people saying don't withdraw the funds until all channels are closed because of anchor something. Can't recall the wording. Would I be able to transfer most of it and leave just a few hundred thousands sats or something in the wallet?

**Reply by guggero** [reply]
Yes, depending on whether it is an anchor channel type, you might need some sats to pull the funds. But a few hundred thousand sats should be enough for that, you can move the rest to another wallet.

**Reply by Cmac0519** [reply]
I am trying to run the tool but keep getting cannot execute binary file: Exec format error

**Reply by guggero** [reply]
Download it for the right architecture/operating system.

**Comment by caiomsbr** [comment]
The new command works flawlessly!!! Thank you so much @guggero!! The might be a horde of nodes with the same problem out there! Now they'll find the solution here!! I wont be opening channels with CLN for the time being!

**Comment by Crypt-iQ** [comment]
Can we close this issue to remove it from the issue tracker and convert it to a discussion?
